cpp/dolfinx/io/XDMFFile.cpp

```diff
@@ -336,8 +337,29 @@ XDMFFile::read_meshtags(const mesh::Mesh<double>& mesh, std::string name,
 
   const auto [entities, eshape] = read_topology_data(name, xpath);
 
-  pugi::xml_node values_data_node
-      = grid_node.child("Attribute").child("DataItem");
+  pugi::xml_node attribute_node = grid_node.child("Attribute");
+  pugi::xml_node values_data_node = attribute_node.child("DataItem");
+  if (not attribute_label.empty())
+  {
+    bool found = false;
+    for (; attribute_node;
```

Comment:
This looks weird and is hard to read. Should it be `while` loop?

Reply:
Done